python/ffsim/states/slater.py

```diff
@@ -0,0 +1,353 @@
+from __future__ import annotations
+
+import numpy as np
+import scipy.linalg
+
+
+def sample_slater(
+    rdm: np.ndarray | tuple[np.ndarray, np.ndarray],
+    chain_length: int,
+    n_chains: int,
+    n_particles_to_move: int = 1,
+    seed=None,
+) -> np.ndarray:
+    """
+    Collect samples of electronic configurations from a Slater determinant defined
+    by its one-body reduced density matrix (RDM). The sampler uses the
+    Metropolis-Hastings method.
+
+    Args:
+        rdm: The one-body reduced density matrix that defines the Slater determinant
+            This is either a single Numpy array specifying the 1-RDM of a
+            spin-polarized system, or a pair of Numpy arrays where each element
+            of the pair contains the 1-RDM for each spin sector.
+        chain_length: Length of each sampling chain.
+        n_chains: Number of chains to run. The total number of samples is given
+            by (``chain_length`` + 1) * ``n_chains``.
+        n_particles_to_move: At each step of the chain, the maximum number of
+            electrons that are moved into a different orbital.
+        seed: Seed for the random number generator.
+
+    Returns:
+        A 2D Numpy array with samples of electronic configurations.
+        Each row is a sample.
+    """
+
+    if isinstance(rdm, np.ndarray) and rdm.ndim == 2:
+        # spinless case
+        n = round(np.real(np.sum(np.diag(rdm))))
+        norb = rdm.shape[0]
```

Comment:
```suggestion
        norb, _ = rdm.shape
```
This has the advantage that it will throw an error on an improperly shaped input.